src/messages/MessageSimilarity.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "Application.hpp"
+#include "controllers/accounts/AccountController.hpp"
+#include "messages/Message.hpp"
+#include "providers/twitch/TwitchAccount.hpp"
+#include "singletons/Settings.hpp"
+
+#include <ranges>
+
+namespace chatterino::similarity::detail {
+
+float relativeSimilarity(QStringView str1, QStringView str2);
+
+float inMessages(const MessagePtr &msg,
```

Comment:
move to source file

Reply:
That's not possible as it takes any range (e.g. vector, limitedqueue, map, etc)

Reply:
We only use 2 types right now (limitedqueue & vector), if we explicitly instantiate those in the source file, the two detail functions can now both be anonymous in the source file instead.
Less flexibility, but given what we support now I think it's ok. If we start supporting a bunch of other types of random range types, we can reconsider it

Reply:
Sure